target/cheri-common/op_helper_cheri_common.c

```diff
@@ -253,16 +344,78 @@ void CHERI_HELPER_IMPL(ccleartag(CPUArchState *env, uint32_t cd, uint32_t cb))
     update_capreg(env, cd, &result);
 }
 
-void CHERI_HELPER_IMPL(cjalr(CPUArchState *env, uint32_t cd, uint32_t cb,
-                             target_ulong offset, target_ulong link_pc))
+void cheri_jump_and_link(CPUArchState *env, const cap_register_t *target,
+                         target_ulong addr, uint32_t link_reg,
+                         target_ulong link_pc, uint32_t cjalr_flags)
+{
+    // FIXME: I am concerned that morello will want to jump to sealed things and
```

Comment:
Are these known now? (And I believe it does, right?)

Reply:
It does, and I think it now gets it right. Comment should be removed.